samples/videoplayer/src/main/kotlin/Queue.kt

```diff
@@ -14,10 +14,10 @@
  * limitations under the License.
  */
 
-class Queue<T>(val maxSize: Int, val none: T) {
-    val array = Array<T>(maxSize, { _ -> none})
-    var head = 0
-    var tail = 0
+class Queue<T>(val maxSize: Int) {
```

Comment:
Эти изменения вряд ли что-то улучшают.

Reply:
This change is to make this `Queue` impl behave like any other idiomatic Kotlin collection (like ArrayList, HashSet, etc). Previous impl of `Queue` forced the user to use nullable types.

For example, `videoQueue` was declared as `Queue<VideoFrame?>` while, in fact, `null` frame were never pushed to queue, so the type was misleading to anyone reading the code. I, frankly, spend some time figuring out why the element type is nullable, since I could not find any code that adds a null frame into this queue.

With this change it is used with non-null types. `videoQueue` is now declared as `Queue<VideoFrame>`. It behaves just like all other Kotlin collections now.